src/qt/pivx/settings/settingsdisplayoptionswidget.cpp

```diff
@@ -144,7 +144,7 @@ void SettingsDisplayOptionsWidget::onResetClicked()
 void SettingsDisplayOptionsWidget::setMapper(QDataWidgetMapper *mapper)
 {
     mapper->addMapping(ui->comboBoxDigits, OptionsModel::Digits);
-    mapper->addMapping(ui->comboBoxLanguage, OptionsModel::Language);
+    mapper->addMapping(ui->comboBoxLanguage, OptionsModel::Language, "currentData");
```

Comment:
This seems to prevent the comboBox from automatically selecting the used language (after restart or after initial welcome screen) but at least fixes the issue at hand. So, can be fixed in a successive PR.